framework/include/csg/CSGBase.h

```diff
@@ -0,0 +1,538 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "CSGSurfaceList.h"
+#include "CSGPlane.h"
+#include "CSGSphere.h"
+#include "CSGXCylinder.h"
+#include "CSGYCylinder.h"
+#include "CSGZCylinder.h"
+#include "CSGRegion.h"
+#include "CSGCellList.h"
+#include "CSGUniverseList.h"
+#include "nlohmann/json.h"
```

Comment:
```suggestion
#include "CSGSurfaceList.h"
#include "CSGRegion.h"
#include "CSGCellList.h"
#include "CSGUniverseList.h"
#include "nlohmann/json.h"
```
We should be able to avoid referring to any specific surface type, out of modularity. 
I think making a utils will help here, notably to make sure a downstream app inventing a new surface type can add it to the list